lib/libyear_bundler/options.rb

```diff
@@ -0,0 +1,55 @@
+require 'optparse'
+require 'libyear_bundler/version'
+
+module LibyearBundler
+  # Uses OptionParser from Ruby's stdlib to hand command-line arguments
+  class Options
+    def initialize(argv)
+      @argv = argv
+      @options = {}
+      @optparser = OptionParser.new do |opts|
+        opts.banner = "Usage: libyear-bundler [Gemfile ...] [options]\n" \
+          'https://github.com/jaredbeck/libyear-bundler/'
+        opts.default_argv = '--libyears'
+        opts.program_name = 'libyear-bundler'
+        opts.version = ::LibyearBundler::VERSION
+
+        opts.on_head('-h', '--help', 'Prints this help') do
+          puts opts
+          exit
+        end
+
+        opts.on('--all', 'Calculate all metrics') do
+          @options[:libyears?] = true
+          @options[:releases?] = true
+          @options[:versions?] = true
+        end
+
+        opts.on('--libyears', '[default] Calculate libyears out-of-date') do
+          @options[:libyears?] = true
+        end
+
+        opts.on('--releases', 'Calculate number of releases out-of-date') do
+          @options[:releases?] = true
+        end
+
+        opts.on('--versions', 'Calculate major, minor, and patch versions out-of-date') do
+          @options[:versions?] = true
+        end
+
+        opts.on('--grand-total', 'Return value for given metric(s)') do
+          @options[:grand_total?] = true
+        end
+      end
+    end
+
+    def parse!
+      @optparser.parse!(@argv)
+      @options
+    rescue OptionParser::InvalidOption => e
+      warn e
+      warn @optparser.help
+      exit 1
```

Comment:
All exit codes should be unique and represented by a constant like `::LibyearBundler::E_INVALID_CLI_USAGE`